core/tauri/src/updater/core.rs

```diff
@@ -661,21 +670,21 @@ fn copy_files_and_run<R: Read + Seek>(archive_buffer: R, extract_path: &Path) ->
   let mut extractor =
     Extract::from_cursor(archive_buffer, ArchiveFormat::Tar(Some(Compression::Gz)));
 
-  extractor.with_files(|entry| {
-    let path = entry.path()?;
-    if path.extension() == Some(OsStr::new("AppImage")) {
-      // if something went wrong during the extraction, we should restore previous app
-      if let Err(err) = entry.extract(extract_path) {
-        Move::from_source(tmp_app_image).to_dest(extract_path)?;
-        return Err(crate::api::Error::Extract(err.to_string()));
+  extractor
+    .with_files(|entry| {
+      let path = entry.path()?;
+      if path.extension() == Some(OsStr::new("AppImage")) {
+        // if something went wrong during the extraction, we should restore previous app
+        if let Err(err) = entry.extract(extract_path) {
+          Move::from_source(tmp_app_image).to_dest(extract_path)?;
+          return Err(crate::api::Error::Extract(err.to_string()));
+        }
```

Comment:
Since you found a nice option to compare the mountpoints, the ultimate fallback imo should be to avoid move, and copy across the filesystem boundaries instead.

Reply:
That is, if none of the 3 dir options appear to be on the same mount point.

Reply:
> well I didn't want to change the logic that much specially since it was audited already. I think if we fallback to the current appimage directory we're safe.

Got it, sure that makes sense. Because I would _expect_ a sibling to the appimage to be the same file system.
Though we should still check that (previous comment).

Reply:
Yeah I did that though it sounds unlikely that it would happen.